File: SYP.Web/Modules/ServerTypes/Products/BrandsEditorForm.ts
import { BooleanEditor, DateEditor, ImageUploadEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface BrandsEditorForm {
    Name: StringEditor;
    Description: StringEditor;
    Logo: ImageUploadEditor;
    IsActive: BooleanEditor;
    InsertDate: DateEditor;
    InsertUserId: IntegerEditor;
    UpdateDate: DateEditor;
    UpdateUserId: IntegerEditor;
}

export class BrandsEditorForm extends PrefixedContext {
    static readonly formKey = 'Products.BrandsEditor';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BrandsEditorForm.init) {
            BrandsEditorForm.init = true;

            var w0 = StringEditor;
            var w1 = ImageUploadEditor;
            var w2 = BooleanEditor;
            var w3 = DateEditor;
            var w4 = IntegerEditor;

            initFormType(BrandsEditorForm, [
                'Name', w0,
                'Description', w0,
                'Logo', w1,
                'IsActive', w2,
                'InsertDate', w3,
                'InsertUserId', w4,
                'UpdateDate', w3,
                'UpdateUserId', w4
            ]);
        }
    }
}

File: SYP.Web/Modules/Products/Brands/Editor/BrandsEditorForm.cs
namespace SYP.Products.Forms;

[FormScript("Products.BrandsEditor")]
[BasedOnRow(typeof(BrandsRow), CheckNames = true)]
public class BrandsEditorForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public string Logo { get; set; }
    public bool IsActive { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
}